### ISSUE TITLE
Error: Invalid or corrupt jarfile spongevanilla-1.12.2-7.3.1-RC391.jar

### ISSUE THREAD

**Issue by vorburger** [OASIS-learn-study/minecraft-storeys-maker#351]
While looking into #349, I've hit this, which is a fun new one I hadn't seen before, while running our current `./test`:

```
(...)
+ /start
[init] Running as uid=0 gid=0 with /data as 'drwxrwxr-x. 9 0 0 4096 Apr  8 21:36 /data'
[init] Resolved version given LATEST into 1.18.2 and major version 1.18
[init] Resolving type given SPONGEVANILLA
[init] Choosing branch for Sponge
[init] Choosing Version for Sponge
[init] Copying any mods over...
[init] Skipping setup of server.properties
[init] Updating ops
[init] 73551f35-7acb-45c0-bc65-8083c53eec69 already present in ops.json, skipping
[mc-image-helper] 17:58:36.247 ERROR : 'get' command failed: Failed to download https://playerdb.co/api/player/minecraft/fb79c1a9-72bd-34dc-9775-c2ac868d7ccf: status code: 500, reason phrase: Internal Server Error
[init] Checking for JSON files.
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
Error: Invalid or corrupt jarfile spongevanilla-1.12.2-7.3.1-RC391.jar
2022-04-11T17:58:37.873Z        WARN    mc-server-runner        sub-process failed      {"exitCode": 1}
2022-04-11T17:58:37.873Z        INFO    mc-server-runner        Done
```

**Comment by vorburger** [comment]
`trash minecraft-server-test-data/spongevanilla-1.12.2-7.3.1-RC391.jar` made the above go away. Then it does:

```
+ /start
[init] Running as uid=0 gid=0 with /data as 'drwxrwxr-x. 9 0 0 4096 Apr 11 18:01 /data'
[init] Resolved version given LATEST into 1.18.2 and major version 1.18
[init] Resolving type given SPONGEVANILLA
[init] Choosing branch for Sponge
[init] Choosing Version for Sponge
[init] Downloading spongevanilla-1.12.2-7.3.1-RC391.jar ...
[init] Copying any mods over...
[init] Skipping setup of server.properties
[init] Updating ops
[init] 73551f35-7acb-45c0-bc65-8083c53eec69 already present in ops.json, skipping
[mc-image-helper] 18:01:57.601 ERROR : 'get' command failed: Failed to download https://playerdb.co/api/player/minecraft/fb79c1a9-72bd-34dc-9775-c2ac868d7ccf: status code: 500, reason phrase: Internal Server Error
[init] Checking for JSON files.
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
Exception in thread "main" java.lang.NoClassDefFoundError: net/minecraft/launchwrapper/Launch
        at org.spongepowered.server.launch.VanillaServerMain.main(VanillaServerMain.java:126)
        at org.spongepowered.server.launch.VersionCheckingMain.main(VersionCheckingMain.java:38)
Caused by: java.lang.ClassNotFoundException: net.minecraft.launchwrapper.Launch
        at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 2 more
2022-04-11T18:01:59.195Z        WARN    mc-server-runner        sub-process failed      {"exitCode": 1}
2022-04-11T18:01:59.196Z        INFO    mc-server-runner        Done
```

What's going on?

**Comment by vorburger** [comment]
`trash minecraft-server-test-data/minecraft_server.1.12.2.jar` did NOT fix above.

`./clean` (which "wipes" `minecraft-server-test-data`; or not quite, but it "resets" it to a clean state) "fixed" it (now seeing #349).
